libraries/WiFi/src/WiFiMulti.cpp

```diff
@@ -85,38 +85,42 @@ bool WiFiMulti::addAP(const char* ssid, const char *passphrase)
     return true;
 }
 
-uint8_t WiFiMulti::run(uint32_t connectTimeout)
+uint8_t WiFiMulti::run(uint32_t connectTimeout, bool scanHidden)
 {
     int8_t scanResult;
     uint8_t status = WiFi.status();
+    int8_t ci = 0;
+
     if(status == WL_CONNECTED) {
         for(uint32_t x = 0; x < APlist.size(); x++) {
             if(WiFi.SSID()==APlist[x].ssid) {
                 return status;
             }
         }
-        WiFi.disconnect(false,false);
-        delay(10);
+	
+	WiFi.persistent(false);
```

Comment:
This looks hacky, what is its purpose?

Reply:
The standard WiFiMulti code will scan for all SSID it can see, and connected you to one that you have requested if it is available... It will show you some information about the hidden SSID, BUT! if the SSID your requesting is hidden, it will NOT connect you to that SSID. The logic in the code is wrong.

The modification we made to WiFiMulti in the pull request above, will connect you to a hidden SSID if it is available....

Reply:
The lower level implementation details of WiFi.disconnect() are not available for me to examine and determine why the delay is required. Through numerous experiments on a number of secured, unsecured, hidden, and visible SSIDs the delay creates a high degree of success when trying subsequent network connections. The delay put in place when connecting to hidden SSID's was also determined experimentally and results in successful connections almost every time. Removing the delay causes the connections to fail due to the asynchronous nature of the lower level wifi drivers.

Reply:
This sounds like an esp32 equivalent to https://github.com/esp8266/Arduino/issues/4372
I have not encountered this issue on esp32, do you have demo code. I suggest the actual problem. This comment however was specifically aimed at `WiFi.persistent(false);`

Reply:
My esp32 is not responding